File: content-solutions/backtracking/seat-orders.js
function solution(names) {
  const n = names.length;
  const used = new Array(n).fill(false);
  const path = [];
  const result = [];
  function place() {
    if (path.length === n) {
      result.push([...path]);
      return;
    }
    for (let i = 0; i < n; i += 1) {
      if (!used[i]) {
        used[i] = true;
        path.push(names[i]);
        place();
        path.pop();
        used[i] = false;
      }
    }
  }
  place();
  return result;
}

File: content-solutions/backtracking/seat-orders.py
def solution(names):
    n = len(names)
    used = [False] * n
    path = []
    result = []

    def place():
        if len(path) == n:
            result.append(path[:])
            return
        for i in range(n):
            if not used[i]:
                used[i] = True
                path.append(names[i])
                place()
                path.pop()
                used[i] = False

    place()
    return result
